python/pyspark/sql/readwriter.py

```diff
@@ -151,8 +173,10 @@ def schema(self, schema: Union[StructType, str]) -> "DataFrameReader":
         if isinstance(schema, StructType):
             jschema = spark._jsparkSession.parseDataType(schema.json())
             self._jreader = self._jreader.schema(jschema)
+            self._schema = schema
```

Comment:
why do we put it in both if and else instead of putting it after if-else only once?

Reply:
I got a Mypy error. I will refactor the way to store this info: [SPARK-45600](https://issues.apache.org/jira/browse/SPARK-45600)